### PR TITLE
ASM-6512 Adding manifests files

### PR DESCRIPTION
**Pull request by hribab**

### CODE REVIEW

File: lib/puppet/type/transport.rb [hunk 1]
```diff
@@ -14,14 +14,14 @@
   newparam(:server) do
   end
 
-  newparam(:scope) do
-    defaultto 0
-    newvalues(0,1,2)
-  end
+#  newparam(:scope) do
```
Comment (sushilrai) [inline lib/puppet/type/transport.rb]: Is this intentional to remove scope from the type definition ? and add commented code

File: lib/puppet/type/transport.rb [hunk 2]
```diff
@@ -14,14 +14,14 @@
...
+#    defaultto 0
+#    newvalues(0,1,2)
+#  end
 
-  newparam(:cli_path) do
-    defaultto "/opt/Navisphere/bin/naviseccli"
-  end
+#  newparam(:cli_path) do
+#    defaultto "/opt/Navisphere/bin/naviseccli"
+#  end
```
Comment (sushilrai) [inline lib/puppet/type/transport.rb]: same as previous

**Comment by sushilrai** [comment]
@hribab I don't see the transport changes

File: manifests/createstoragegroup.pp [hunk 3]
```diff
@@ -0,0 +1,15 @@
+define vnx::createstoragegroup(
```
Comment (sushilrai) [inline manifests/createstoragegroup.pp]: Indentation is incorrect. Need to add puppet-lint validation in the travis tests for this module

Comment (sushilrai) [inline manifests/createstoragegroup.pp]: Creating hosts from initiators is missing from the manifest. Is this not required for VMware ?

Comment (hribab) [inline manifests/createstoragegroup.pp]: @sushilrai once we add host to zone in switch, that host name will appear in storage facts immediately. We can choose either we go with hbauid or host_name. 
define vnx::createstoragegroup(
$hbauid="",
$sgname="",
$sp="",
$spport=""
){
vnx_storagegroup{$name:
hbauid=>$hbauid,
sg_name=>$sgname,
sp=>$sp,
spport=>$spport,
transport=>$transport
}
hbauid is WWNN::WWPN

I need to test which method works better for multi host, right now host_names working good with single host

File: lib/puppet_x/puppetlabs/transport/emc_vnx.rb [hunk 4]
```diff
@@ -90,8 +90,8 @@ def initial_cli()
           option[:password] = res.provider.send(:password)
           option[:host_addr] = res.provider.send(:server)
           option[:array] = res.provider.send(:name)
-          option[:scope] = res.provider.send(:scope)
-          option[:cli_path] = res.provider.send(:cli_path)
+          option[:scope] = 0 #res.provider.send(:scope)
+          option[:cli_path] = '/opt/Navisphere/bin/naviseccli'#res.provider.send(:cli_path)
```
Comment (gavin-scott) [inline lib/puppet_x/puppetlabs/transport/emc_vnx.rb]: I don't think we should start hard-coding these, see if you can get it to work in the transport.

File: manifests/createstoragegroup.pp [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+vnx_storagegroup{"$name":
+sg_name=>"$sgname",
+luns=>$luns,
+host_name=>$host_name,
+addonly=>"true",
+ensure=>"present",
+transport=>"$transport",
+}
+}
```
Comment (gavin-scott) [inline manifests/createstoragegroup.pp]: Bad formatting, run `puppet-lint` on this.

File: manifests/createvolume.pp [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+){
+vnx_lun{"$name":
+  lun_name=>$name,
+  type=>"$type",
+  capacity=>"$capacity",
+  default_owner=>"$sp",
+  pool_name=>"$poolName",
+  transport=>"$transport",
+}
+}
```
Comment (gavin-scott) [inline manifests/createvolume.pp]: `puppet-lint`

File: manifests/createstoragegroup.pp [hunk 7]
```diff
@@ -0,0 +1,14 @@
+define vnx::createstoragegroup(
+  $sgname,
+  $host_name,
+  $luns={},
+){
+vnx_storagegroup{$name:
```
Comment (gavin-scott) [inline manifests/createstoragegroup.pp]: spaces around `{`

File: manifests/createstoragegroup.pp [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+ensure    =>  'present',
```
Comment (gavin-scott) [inline manifests/createstoragegroup.pp]: Indent by two spaces

File: manifests/createvolume.pp [hunk 9]
```diff
@@ -0,0 +1,15 @@
+define vnx::createvolume(
+  $type     = 'NonThin',
+  $capacity = '1',
+  $sp       = 'a',
+  $poolName = 'Pool 0',
+){
+vnx_lun{$name:
```
Comment (gavin-scott) [inline manifests/createvolume.pp]: spaces around `{`

Comment (gavin-scott) [inline manifests/createvolume.pp]: and indent two spaces

**Comment by gavin-scott** [comment]
The transport changes look good AFAICT

**Comment by sushilrai** [comment]
Merging PR. Will take care of missing formatting issues in separate PR

**Pull request merged**